tasks/nc_installation.yml

```
@@ -175,3 +175,33 @@
     - config
     - themes
     - updater
+
+- name: "[NC] - Disable Nextcloud apps"
+  when: nextcloud_disable_apps is defined
+  block:
+    - name: "[NC apps] - Lists the number of apps available in the instance."
+      ansible.builtin.command: php occ app:list --output=json_pretty --no-warnings
+      args:
+        chdir: "{{ nextcloud_webroot }}"
+      become_user: "{{ nextcloud_websrv_user }}"
+      become: true
+      become_flags: "{{ ansible_become_flags | default(omit) }}"
+      changed_when: false
+      check_mode: false
+      register: nc_apps_list
+
+    - name: "[NC apps] - Convert list to yaml."
+      ansible.builtin.set_fact:
+        nc_available_apps: "{{ nc_apps_list.stdout | from_json }}"
+
+    - name: "[NC apps] - Disable apps"
+      ansible.builtin.command: php occ app:disable "{{ item }}"
+      become_user: "{{ nextcloud_websrv_user }}"
+      become_flags: "{{ ansible_become_flags | default(omit) }}"
+      become: true
+      when:
+        - item in nc_available_apps.enabled
+      changed_when:
+        - "'enabled' in stdout"
+        - "'No such app' not in stdout"
+      with_items: "{{ nextcloud_disable_apps }}"
```

As described in my previous comments, this only task _should_ be enough to do the disable step
```suggestion
    - name: "[NC apps] - Disable apps"
      ansible.builtin.command: php occ app:disable "{{ item }}"
      become_user: "{{ nextcloud_websrv_user }}"
      become_flags: "{{ ansible_become_flags | default(omit) }}"
      become: true
      register: _occ_app_disable_result
      changed_when:
        - "'disabled' in _occ_app_disable_result.stdout"
        - "'No such app' not in _occ_app_disable_result.stdout"
      with_items: "{{ nextcloud_disable_apps }}"
```